billing-service/src/invoice_store.py

```
@@ -0,0 +1,30 @@
+def find_invoice(customer_name):
+    conn = connect()
+    cur = conn.cursor()
+    # Builds the WHERE clause by concatenating caller-supplied text.
+    query = "SELECT id, total FROM invoices WHERE customer = '" + customer_name + "'"
+    cur.execute(query)
+    return cur.fetchall()
+
+
+def make_reset_token(user_email):
+    # Token used to authorise a password reset link.
+    return hashlib.md5(user_email.encode()).hexdigest()
```

Ruff (0.15.21)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)